sigstore/_internal/merkle.py

```diff
@@ -133,3 +134,33 @@ def verify_merkle_inclusion(entry: LogEntry) -> None:
             f"Inclusion proof contains invalid root hash: expected {inclusion_proof}, calculated "
             f"{calc_hash}"
         )
+
+
+def verify_checkpoint(client: RekorClient, entry: LogEntry) -> None:
+    """
+    Verify the inclusion proof's checkpoint.
+    """
+
+    inclusion_proof = entry.inclusion_proof
+    if inclusion_proof is None:
+        raise InvalidInclusionProofError("Rekor entry has no inclusion proof")
+    if inclusion_proof.checkpoint is None:
+        return
```

Comment:
Shouldn't this be an error case? We probably shouldn't silently accept inclusion proofs that don't contain a checkpoint.